torch/_prims_common/__init__.py

```diff
@@ -1784,12 +1784,11 @@ def get_torch_state_as_tuple(fake_mode=None):
                 offset = fake_mode.from_tensor(offset)
             return seed, offset
 
-        rng_state = torch.cuda.get_rng_state()
+        seed = torch.tensor(torch.cuda.initial_seed())
+        offset = torch.tensor(torch.cuda._get_rng_state_offset())
```

Comment:
These lines are the ones I mentioned in our 1:1, @anijain2305, instead of allocating a real tensor and then fakeifying it, you should `with fake_mode` and then allocate the tensor within there. In fact, you do not even need the real value, because these are fake tensors! The real value should get read in at some other point.

This is technically a preexisting problem but I think this bit can be simplified a lot if you do it this way.

Reply:
Thanks for noticing this. Changed!